### ISSUE TITLE
SITL no longer compiles under macOS

### ISSUE THREAD

**Issue by charlie-mtz** [ArduPilot/ardupilot#8592]
#### Issue details
SITL no longer compiles under macOS.
Commit f986f1366ff55859bb96764a09fc3e28d83ec776 adds SOCK_CLOEXEC option to the socket type. This is not standard unix and doesn't work on the Mac.

A possible solution to this would be to do it the old way.
Instead of:
```
fd = socket(AF_INET, SOCK_STREAM | SOCK_CLOEXEC, 0);
```
do:
```
fd = socket(AF_INET, SOCK_STREAM, 0);
fcntl(fd, O_CLOEXEC);
```

#### Version
master at commit f986f1366ff55859bb96764a09fc3e28d83ec776
#### Platform
[  ] All
[  ] AntennaTracker
[x] Copter
[  ] Plane
[  ] Rover
[  ] Submarine

#### Airframe type
SITL

#### Hardware type
_What autopilot hardware was used? (pixhawk, pixracer, PX4FMU etc)_

#### Logs
```
[368/493] Compiling libraries/AP_HAL_SITL/UARTDriver.cpp
../../libraries/AP_HAL_SITL/UARTDriver.cpp: In member function 'void HALSITL::UARTDriver::_tcp_start_connection(uint16_t, bool)':
../../libraries/AP_HAL_SITL/UARTDriver.cpp:225:52: error: 'SOCK_CLOEXEC' was not declared in this scope
         _listen_fd = socket(AF_INET, SOCK_STREAM | SOCK_CLOEXEC, 0);
                                                    ^~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
```

**Comment by charlie-mtz** [comment]
Ok.

**Comment by lvale** [comment]
http://man7.org/linux/man-pages/man2/socket.2.html

**Comment by charlie-mtz** [comment]
Submitted pull request https://github.com/ArduPilot/ardupilot/pull/8593.

**Comment by OXINARF** [comment]
Fixed via #8593.

**Comment by barracuda156** [comment]
Maybe this could work instead: https://fjordtek.com/git/Fincer/openntpd-portable/commit/e3106f50ac730b480f4a9dc05ee02e32d318bef6
